netplan_cli/cli/state.py

```diff
@@ -36,6 +36,101 @@
 
 JSON = Union[Dict[str, 'JSON'], List['JSON'], int, str, float, bool, Type[None]]
 
+DEVICE_TYPES = dict({
+    'bond': 'bond',
+    'bridge': 'bridge',
+    'ether': 'ethernet',
+    'ipgre': 'tunnel',
+    'ip6gre': 'tunnel',
+    'loopback': 'ethernet',
+    'sit': 'tunnel',
+    'tunnel': 'tunnel',
+    'tunnel6': 'tunnel',
+    'wireguard': 'tunnel',
+    'wlan': 'wifi',
+    'wwan': 'modem',
+    'vlan': 'vlan',
+    'vrf': 'vrf',
+    'vxlan': 'tunnel',
+
+    # Netplan netdef types
+    'wifis': 'wifi',
+    'ethernets': 'ethernet',
+    'bridges': 'bridge',
+    'bonds': 'bond',
+    'nm-devices': 'nm-device',
+    'dummy-devices': 'dummy',
+    'modems': 'modem',
+    'vlans': 'vlan',
+    'vrfs': 'vrf',
+    })
+
+
+class DiffJSONEncoder(json.JSONEncoder):
+    def default(self, obj):
+        if isinstance(obj, Route):
+            return obj.to_dict()
+
+        # Shouldn't be reached as the only non-serializable type we have at the moment is Route
+        return json.JSONEncoder.default(self, obj)  # pragma: nocover (only Route requires the encoder)
+
+
+class Route(libnetplan.NetplanRoute):
+    def __init__(self, route: libnetplan.NetplanRoute = None):
+        super().__init__()
+        if route:
+            self.__dict__ = route.__dict__.copy()
+
+    def __str__(self):
+        route = ""
+        if self.to:
+            route = route + self.to
+        if self.via:
+            route = route + f' via {self.via}'
+        if self.type:
+            route = route + f' type {self.type}'
+        if self.scope:
+            route = route + f' scope {self.scope}'
+        if self.from_addr:
+            route = route + f' src {self.from_addr}'
+        if self.metric < self.METRIC_MAX_:
+            route = route + f' metric {self.metric}'
+        if self.table:
+            route = route + f' table {self.table}'
+        return route
+
+    def __eq__(self, route):
```

Comment:
note: This will be interesting, comparing routes is tricky. I'll need to have a closer look into that.